src/axom/primal/operators/winding_number.hpp

```diff
@@ -109,16 +110,19 @@ int winding_number(const Point<T, 2>& q,
 template <typename T>
 int winding_number(const Point<T, 2>& R,
                    const Polygon<T, 2>& P,
+                   bool& isOnEdge,
                    bool includeBoundary = false,
                    double EPS = 1e-8)
 {
   const int nverts = P.numVertices();
+  isOnEdge = false;
 
   // If the query is a vertex, return a value interpreted
   //  as "inside" by evenodd or nonzero protocols
   if(axom::utilities::isNearlyEqual(P[0][0], R[0], EPS) &&
      axom::utilities::isNearlyEqual(P[0][1], R[1], EPS))
   {
+    isOnEdge = true;
     return includeBoundary;
```

Comment:
Should this return an integer, rather than a `bool`?
If so, would something like the following be more appropriate/clear? (here and elsewhere in this function).
```suggestion
    return includeBoundary ? 1 : 0;
```

... or am I misunderstanding something?

Reply:
This is relying on the automatic conversion from bool to integer here, yes. Your version is certainly more clear semantically, probably even more than a cast from one type to the other would be.